### ISSUE TITLE
Vulnerability : unsigned JWT wrongly considered valid

### ISSUE THREAD

**Issue by zeludo** [alvarosanchez/grails-spring-security-rest#284]
My application has been pen tested and a critical vulnerability was found concerning the token verification. The plugin is configured for using signed jwt
```
grails.plugin.springsecurity.rest.token.storage.useJwt = true
grails.plugin.springsecurity.rest.token.storage.jwt.useSignedJwt = true
```
 but an attacker would easily be able to bypass signature verification by following these steps :

1. Generate a token
1. Replace JWT header with `{"alg":"none"}` so `eyJhbGciOiJub25lIn0=` in base64
1. Delete the signature part of the token, just leaving the `.` at the end
1. Use this token to authenticate a request

This was done using v1.5.3, haven't tested on v2.

**Comment by r1secure** [comment]
It's really a critical vulnerability. My company is also concerned. Pease, could you look at this issue?

**Comment by zeludo** [comment]
Sorry I forgot to make a PR. See #286

**Comment by alvarosanchez** [comment]
The PR cannot be merged automatically, but I will merge it manually and release a new 1.5.x version.

Thank you very much!

**Comment by r1secure** [comment]
Thanks

**Comment by alvarosanchez** [comment]
Versions `1.5.4` and `2.0.0.BUILD-SNAPSHOT` have been just published

**Comment by immutability** [comment]
Hello! We recently ran into this issue so we just did a local build of the plugin to package with our application, but I wanted to check if you're planning to do an "official" build of 2.0.0 anytime soon (that would have the fix)? It looks like 2.0.0.M2 (which is affected) is still the most recent release. Thanks!
